### ISSUE TITLE
optimizer Params setting and Loss backward

### ISSUE THREAD

**Issue by henry-tujia** [fkodom/lora-pytorch#1]
Thank you for the valuable contribution! I've encountered a few issues while training a ResNet-18 model on the CIFAR-10 dataset using the provided code.

### Training Code
```python
from lora_pytorch import LoRA
import torch
import torchvision.datasets as datasets
import torchvision.transforms as transforms
from torchvision.models import resnet18
import torch.nn as nn
from torch.optim import Adam
from tqdm import tqdm

# Setup and data preprocessing
transform = transforms.Compose([
    transforms.ToTensor(),
    transforms.Normalize((0.1307,), (0.3081,))
])

train_dataset = datasets.CIFAR10(root="./datasets", train=True, download=True, transform=transform)
test_dataset = datasets.CIFAR10(root="./datasets", train=False, download=True, transform=transform)

train_loader = torch.utils.data.DataLoader(train_dataset, batch_size=10, shuffle=True)
test_loader = torch.utils.data.DataLoader(test_dataset, batch_size=10, shuffle=True)

device = torch.device("cuda" if torch.cuda.is_available() else "cpu")

# Model initialization and optimization
origin_model = resnet18(num_classes=10).to(device)
model = LoRA.from_module(origin_model, rank=5)
optimizer = Adam(model.parameters(), lr=0.001)

# Training function
def train_model(train_loader, model, epochs, total_iterations_limit=None):
    criterion = nn.CrossEntropyLoss()
    for epoch in range(epochs):
        model.train()
        running_loss = 0.0

        for i, data in enumerate(tqdm(train_loader, desc=f"Epoch {epoch+1}")):
            inputs, labels = data[0].to(device), data[1].to(device)
            optimizer.zero_grad()
            outputs = model(inputs)
            loss = criterion(outputs, labels)
            loss.backward()
            optimizer.step()
            running_loss += loss.item()

            if total_iterations_limit and i + 1 >= total_iterations_limit:
                break

# Testing function
def test_model():
    model.eval()
    correct = 0
    total = 0
    wrong_counts = torch.zeros(10)

    with torch.no_grad():
        for data in tqdm(test_loader, desc="Testing"):
            inputs, labels = data[0].to(device), data[1].to(device)
            outputs = origin_model(inputs)
            _, predicted = torch.max(outputs.data, 1)
            total += labels.size(0)
            correct += (predicted == labels).sum().item()
            wrong_counts[labels[predicted != labels]] += 1

    accuracy = correct / total
    print(f"Test Accuracy: {accuracy:.3f}")
    for i, count in enumerate(wrong_counts):
        print(f"Class {i} misclassifications: {count}")

# Execute training and testing
train_model(train_loader, model, epochs=10)
test_model()
```
### Concerns
1. The `model` object appears to have no parameters when I attempt to retrieve them with `model.parameters()`, which returns `None`. This is due to the `LoRA` module being `None`. I'm unsure whether it's correct to pass `model` to the optimizer instead of the `origin_model`.
2. When attempting to run `loss.backward()` with the `origin_model` in the optimizer, I receive an error: "element 0 of tensors does not require grad and does not have a grad_fn". I'm uncertain if I'm using the repository correctly. Could you please advise?

**Comment by bvanbreugel** [comment]
I ran into the same issue. I changed ``self.parameters`` to return the internal model, and this seems to work fine. I created a fork, https://github.com/bvanbreugel/lora-pytorch, which runs fine (though haven't properly tested it yet)

**Comment by henry-tujia** [comment]
> I ran into the same issue. I changed `self.parameters` to return the internal model, and this seems to work fine. I created a fork, https://github.com/bvanbreugel/lora-pytorch, which runs fine (though haven't properly tested it yet)

Wow! Thanks for that. I will check it!
